spec/linkedList.js

```diff
@@ -0,0 +1,213 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedList'
+
+chai.use(chaiChange)
+
+describe.only('LinkedList', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('insert()', () => {
+    it('adds a new node to the linked list when empty', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      expect(myLinkedList.size()).to.eql(1)
+    })
+    it('adds new node to the end if list is not empty', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      myLinkedList.insert("Beelzebub")
+      expect(myLinkedList.size()).to.eql(2)
+      expect(myLinkedList.getHeadNode()).to.eql({data: "Azazel", next:
+        {data: "Beelzebub", next: null}})
+    })
+  })
+
+  context('getHeadNode()', () => {
+    it('returns the data for the head node of the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      myLinkedList.insert("Beelzebub")
+      expect(myLinkedList.getHeadNode()).to.eql({data: "Azazel", next:
+        {data: "Beelzebub", next: null}})
+    })
+    it('returns null if list is empty', () => {
+      const myLinkedList = new LinkedList()
+      expect(myLinkedList.getHeadNode()).to.eql(null)
+    })
+  })
+
+  context('size()', () => {
+    it('returns the total number of nodes in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      myLinkedList.insert("Beelzebub")
+      expect(myLinkedList.size()).to.eql(2)
+    })
+    it('returns 0 if there are no nodes in the list', () => {
+      const myLinkedList = new LinkedList()
+      expect(myLinkedList.size()).to.eql(null)
+    })
+  })
+
+  context('isEmpty()', () => {
+    it('returns true if there are no nodes in the list', () => {
+      const myLinkedList = new LinkedList()
+      expect(myLinkedList.isEmpty()).to.eql(true)
+    })
+    it('returns false if there are nodes in the list', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      expect(myLinkedList.isEmpty()).to.eql(false)
+    })
+  })
+
+  context('clear()', () => {
+    it('Resets the list to empty', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("Azazel")
+      expect(myLinkedList.size()).to.eql(1)
+      myLinkedList.clear()
+      expect(myLinkedList.size()).to.eql(null)
```

Comment:
why is the size null? does size initially start off at null? it makes more sense to me that it would start off at 0